src/Workspaces/Core/Portable/Workspace/Solution/Document.cs

```diff
@@ -257,6 +262,30 @@ public bool TryGetSemanticModel([NotNullWhen(returnValue: true)] out SemanticMod
             return _model != null && _model.TryGetTarget(out semanticModel);
         }
 
+        /// <summary>
+        /// Gets the current nullable disabled semantic model for this document if the model is already computed and still cached.
+        /// In almost all cases, you should call <see cref="GetSemanticModelAsync"/>, which will compute the semantic model
+        /// if necessary.
+        /// </summary>
+        public bool TryGetNullableDisabledSemanticModel([NotNullWhen(returnValue: true)] out SemanticModel? semanticModel)
+        {
+            semanticModel = null;
+            return _nullableDisabledModel != null && _nullableDisabledModel.TryGetTarget(out semanticModel);
+        }
+
+        /// <summary>
+        /// Gets the nullable disbaled semantic model for this document asynchronously.
+        /// </summary>
+        /// <returns>
+        /// The returned <see cref="SemanticModel"/> may be <see langword="null"/> if <see
+        /// cref="SupportsSemanticModel"/> returns <see langword="false"/>. This function will
+        /// return the same value if called multiple times.
+        /// </returns>
+        public async Task<SemanticModel?> GetNullableDisabledSemanticModelAsync(CancellationToken cancellationToken = default)
```

Comment:
If we have SemanticModelOptions as a type from the compiler, I'd imagine this would just be an overload of GetSemanticModel that takes that object as an additional parameter.